src/fixate/_switching.py

```diff
@@ -441,6 +456,83 @@ def _default_update_pins(
         """
         print(pin_updates, trigger_update)
 
+    # This looks bad, maybe we should not type hint this
+    def __class_getitem__(cls, arg: Union[TypeVar, Any]) -> type[VirtualMux[S]]:
```

Comment:
I think it should be `arg: Union[TypeVar, Type]`

Reply:
Technically this is correct and counterintuitively here I think it makes more sense that the type hint here is for what happens at runtime so that you can have useful introspection inside the function.
It seems like the inheritance from Generic is what sets the type hint on this call for tools like mypy